index.js

```diff
@@ -189,3 +169,47 @@ async function invalidation (invalidate, cache, name, self, arg, ctx, info, resu
     onError(err)
   }
 }
+
+async function getResultForMutationSubscription ({ self, arg, ctx, info, originalFieldResolver, onError }) {
+  let result = false
+  if (info.operation && (info.operation.operation === 'mutation' || info.operation.operation === 'subscription')) {
+    try {
+      result = await originalFieldResolver(self, arg, ctx, info)
+    } catch (error) {
+      onError(error)
+      return null
+    }
+  }
+  return result
+}
+
+async function getResultIfSkipDefined ({ self, arg, ctx, info, skip, policy, name, report, originalFieldResolver, onError }) {
+  const result = false
+  let isSkipped = false
+  try {
+    if ((skip && (await skip(self, arg, ctx, info))) ||
+      (policy && policy.skip && (await policy.skip(self, arg, ctx, info)))) isSkipped = true
+  } catch (error) {
+    onError(error)
+    return await originalFieldResolver(self, arg, ctx, info)
+  }
+
+  if (isSkipped) {
+    try {
+      report[name].onSkip()
+      return await originalFieldResolver(self, arg, ctx, info)
+    } catch (error) {
+      onError(error)
+      return null
+    }
```

Comment:
This is not correct and a bug in the previous implementation. We should not catch this error nor invoke onError.